Code/Framework/AzFramework/Platform/Windows/AzFramework/Device/DeviceAttributesCommon_Win.cpp

```diff
@@ -10,37 +10,130 @@
+                classObject->Release();
             }
-            ::RegCloseKey(hKey);
         }
-        else
+
+        if (classObjectEnumerator)
         {
-            AZLOG_WARN(R"(Unable to determine device model attribute because the registry HKEY_LOCAL_MACHINE\HARDWARE\DESCRIPTION\System\BIOS could not be read.)");
+            classObjectEnumerator->Release();
         }
+
+        return result;
+    }
+
+    AZStd::string GetWMIPropertyValue(const AZStd::wstring& propertyName)
+    {
+        AZStd::string propertyValue;
+
+        // NOTE: this query functionality is not optimized for querying multiple values
+        // If it becomes necessary to add more device attributes that must also query WMI
+        // an optimization can be to query all the WMI properties for all attributes in a
+        // single query, store in a static cache so they are available as the other device
+        // attributes are created so they can pull the values from the cache.
+
+        CoInitialize(nullptr);
+
+        // Obtain the initial locator to Windows Management on a particular host computer.
+        IWbemLocator* locator = nullptr;
+        auto hResult = CoCreateInstance(CLSID_WbemLocator, 0, CLSCTX_INPROC_SERVER, IID_IWbemLocator, (LPVOID*)&locator);
+        if (!FAILED(hResult))
+        {
+            // Connect to the root\cimv2 namespace with the current user and obtain pointer pSvc to make IWbemServices calls.
+            IWbemServices* services = nullptr;
+            auto serverPath = SysAllocString(TEXT(R"(ROOT\CIMV2)"));
+            hResult = locator->ConnectServer(serverPath, nullptr, nullptr, 0, NULL, 0, 0, &services);
+            SysFreeString(serverPath);
+
+            if (!FAILED(hResult))
+            {
+                // Set the IWbemServices proxy so that impersonation of the user (client) occurs.
+                hResult = CoSetProxyBlanket( services,
+                    RPC_C_AUTHN_WINNT,
+                    RPC_C_AUTHZ_NONE,
+                    nullptr,
+                    RPC_C_AUTHN_LEVEL_CALL,
+                    RPC_C_IMP_LEVEL_IMPERSONATE,
+                    nullptr,
+                    EOAC_NONE);
+
+                if (!FAILED(hResult))
+                {
+                    propertyValue = QueryWMIProperty(services, propertyName);
+                }
+            }
+
+            if (services)
+            {
+                services->Release();
+            }
+        }
+
+        if (locator)
+        {
+            locator->Release();
+        }
+
+        CoUninitialize();
+
+        return propertyValue;
+    }
+
+    DeviceAttributeDeviceModel::DeviceAttributeDeviceModel()
+    {
+        // Use WMI because the registry HARDWARE\DESCRIPTION\System\BIOS\SystemProductName is not
+        // always available or non-empty and the name does not match what WMI returns for the Model
+        m_value = GetWMIPropertyValue(TEXT("Model"));

```

Comment:
This can pass in the "Model" string directly if the paramter is an `AZStd::string_view`
```suggestion
        // Use WMI because the registry HARDWARE\DESCRIPTION\System\BIOS\SystemProductName is not
        // always available or non-empty and the name does not match what WMI returns for the Model
        m_value = GetWMIPropertyValue("Model");

```